astropy/utils/masked/core.py

```diff
@@ -348,12 +354,10 @@ def _construct_from_dict(self, map):
         return Masked(map.pop('data').data, mask=map.pop('mask', None))
 
 
-class MaskedArraySubclassInfo:
+class MaskedArraySubclassInfo(MaskedInfoBase):
     """Mixin class to create a subclasses such as MaskedQuantityInfo."""
-    # This is used below in __init_subclass__.
-    # TODO: Find some way to ensure that 'serialize_method' of MaskedNDArray
```

Comment:
So, the answer to this `TODO` was to ensure `info` was propagated correctly, which means 'serialize_method' is passed on as well. The system actually worked brilliantly once I realized what was wrong.